src/data/int/gcd.lean

```diff
@@ -302,6 +302,13 @@ begin
   exact int.nat_abs_dvd_iff_dvd.mpr h
 end
 
+lemma gcd_greatest {a b d : ℤ} (hd : 0 ≤ d) (hda : d ∣ a) (hdb : d ∣ b)
+  (he : ∀ e : ℤ, e ∣ a → e ∣ b → e ∣ d) : d = (gcd a b) :=
```

Comment:
Possible generalizations:
* For `gcd_monoid α`, without `hd`, with conclusion `associated d (gcd a b)`
* For `normalized_gcd_monoid α`, without `hd`, with conclusion `gcd a b = normalize d` (similar to `gcd_eq_normalize`)
* For `normalized_gcd_monoid α`, with `hd : normalize d = d` and conclusion `d = gcd a b`

Reply:
I've added the three generalisations for `normalized_gcd_monoid` and `gcd_monoid` to `algebra/gcd_monoid/basic`.